src/components/Gradient/index.js

```diff
@@ -2,91 +2,86 @@ import React, { PureComponent } from "react";
+  OVERLAY_STOPS_NOT_SPOTLIGHT,
+  OVERLAY_SHADOW_DEG,
+  OVERLAY_SHADOW_STOPS,
+  OVERLAY_SHADOW_TOP_DEG,
+  OVERLAY_SHADOW_TOP_HEIGHT,
+  OVERLAY_SHADOW_TOP_STOPS,
+  SPOTLIGHT_DEG,
+  SPOTLIGHT_STOPS
+} from "./constants";
+import { zIndex } from "../../theme";
+import { largeAndUp } from "../../theme/mediaQueries";
+import {
+  linearGradient,
+  responsiveBackgroundImages,
+  responsiveLinearGradient,
+  responsiveLinearGradients
+} from "../../mixins";
 import SpotLight from "./SpotLight";
 import Angle from "./Angle";
 import StyledImageSeo from "../Image/Seo.styles";
 
-const SPOTLIGHT_STOPS = [
-  "rgb(0, 45, 161)",
-  `${themes.global.primary.base} 55%`,
-  "rgb(0, 45, 161)"
-];
-
-const SPOTLIGHT_DEG = {
-  small: "256deg",
-  medium: "260deg",
-  large: "262deg"
-};
-
 export const GradientStyles = styled.span`
-  z-index: 1;
+  z-index: ${zIndex.reset};
   width: 100%;
 
-  background-image: ${props =>
-    `linear-gradient(${props.deg.small}, ${props.stops.join(",")})`};
-  ${mediumAndUp`
-    background-image: ${props =>
-      `linear-gradient(${props.deg.medium}, ${props.stops.join(",")})`};
-  `} ${largeAndUp`
-    background-image: ${props =>
-      `linear-gradient(${props.deg.large}, ${props.stops.join(",")})`};
-  `};
-
   &.gradient--overlay:after,
-  &.gradient--underlay:after {
-    height: 100%;
+  &.gradient--underlay:after,
+  &.gradient--overlay-shadow:after {
     content: "";
     top: 0;
     left: 0;
     right: 0;
     position: absolute;
   }
 
+  &.gradient--overlay:after,
+  &.gradient--underlay:after {
+    height: 100%;
+  }
+
+  &.gradient--overlay-shadow:after {
+    height: ${OVERLAY_SHADOW_TOP_HEIGHT};
+    ${linearGradient({
+      deg: OVERLAY_SHADOW_TOP_DEG,
+      stops: OVERLAY_SHADOW_TOP_STOPS
+    })};
+  }
+
   &.gradient--spotlight {
     ${largeAndUp`box-shadow: inset 0 1px 40px 10px rgba(0, 0, 0, 0.15);`};
   }
 
-  &.gradient--underlay.gradient--spotlight:after {
-    z-index: -1;
-    opacity: 0.8;
-    background-image: ${props =>
-      `linear-gradient(${props.deg.small}, ${props.stops.join(",")})`};
-    ${mediumAndUp`
-      background-image: ${props =>
-        `linear-gradient(${props.deg.medium}, ${props.stops.join(",")})`};
-    `} ${largeAndUp`
-      background-image: ${props =>
-        `linear-gradient(${props.deg.large}, ${props.stops.join(",")})`};
-    `};
+  &:not(.gradient--overlay-shadow),
+  &.gradient--underlay.gradient--spotlight:after,
+  &:not(.gradient--spotlight).gradient--underlay:after {
+    ${responsiveLinearGradients};
   }
 
-  &:not(.gradient--spotlight).gradient--overlay:after {
-    z-index: 2;
-    opacity: 0.4;
-    background-image: linear-gradient(77deg, rgba(0, 0, 0, 0), #000000);
-    ${mediumAndUp`
-        background-image: linear-gradient(82deg, rgba(0, 0, 0, 0), #000000);
-      `};
-
-    ${largeAndUp`
-      background-image: linear-gradient(86deg, rgba(0, 0, 0, 0), #000000);
-      `};
+  &.gradient--overlay-shadow {
+    z-index: inherit;
```

Comment:
do you indent to `inherit` z-index from the parent? or did you mean to "unset" the z-index? in that case `auto` would be the value to to use to avoid  creation of a new stacking context

Reply:
Good catch, the value should be `zIndex.reset` 👍